spec/rubocop/cop/rails/where_missing_spec.rb

```diff
@@ -0,0 +1,283 @@
+
+    it 'registers an offense when using `left_joins(:foo).where(bar: "bar").where(foos: {id: nil})` and ' \
+       'there is a line break after `where(bar: "bar")`' do
+      expect_offense(<<~RUBY)
+        Foo
+          .left_joins(:foo).where(bar: "bar")
+           ^^^^^^^^^^^^^^^^ Use `where.missing(:foo)` instead of `left_joins(:foo).where(foos: { id: nil })`.
+          .where(foos: { id: nil })
+      RUBY
+
+      expect_correction(<<~RUBY)
+        Foo
+          .where.missing(:foo).where(bar: "bar")
+      RUBY
+    end
+
+    it 'registers an offense when using `left_joins(:foo).where(foos: {id: nil})` and there is a line break after ' \
+       '`left_joins.where` and receiver' do
+      expect_offense(<<~RUBY)
+        Foo
+          .left_joins(:foo).where(foos: { id: nil })
+           ^^^^^^^^^^^^^^^^ Use `where.missing(:foo)` instead of `left_joins(:foo).where(foos: { id: nil })`.
+          .where(bar: "bar")
+      RUBY
+
+      expect_correction(<<~RUBY)
+        Foo
+          .where.missing(:foo)
+          .where(bar: "bar")
+      RUBY
+    end
+
+    it "registers an offense when using `left_joins(:foo).where(foos: {id: nil}, bar: 'bar')` " \
+       'with other `left_joins(:foo)`' do
+      expect_offense(<<~RUBY)
+        def test
+          Foo.left_joins(:foo).any?
+
+          do_something
+
+          Foo.left_joins(:foo).where(foos: {id: nil})
+              ^^^^^^^^^^^^^^^^ Use `where.missing(:foo)` instead of `left_joins(:foo).where(foos: { id: nil })`.
+        end
+      RUBY
+
+      expect_correction(<<~RUBY)
+        def test
+          Foo.left_joins(:foo).any?
+
+          do_something
+
+          Foo.where.missing(:foo)
+        end
+      RUBY
+    end
+
+    it "registers an offense when using `left_joins(:foo).where(foos: {id: nil}, bar: 'bar')` " \
+       "with other `left_joins(:foo).where(foos: {id: nil}, bar: 'bar')`" do
+      expect_offense(<<~RUBY)
+        def test
+          Foo.left_joins(:foo).where(foos: {id: nil})
+              ^^^^^^^^^^^^^^^^ Use `where.missing(:foo)` instead of `left_joins(:foo).where(foos: { id: nil })`.
+
+          do_something
+
+          Foo.left_joins(:foo).where(foos: {id: nil})
+              ^^^^^^^^^^^^^^^^ Use `where.missing(:foo)` instead of `left_joins(:foo).where(foos: { id: nil })`.
+        end
+      RUBY
+
+      expect_correction(<<~RUBY)
+        def test
+          Foo.where.missing(:foo)
+
+          do_something
+
+          Foo.where.missing(:foo)
+        end
+      RUBY
+    end
+
+    it 'does not register an offense when not finding missing relationship records' do
+      expect_no_offenses(<<~RUBY)
+        Foo.left_joins(:foo).where(bazs: { id: nil })
+        Foo.left_joins(:foo).where(foos: { name: nil })
+        Foo.left_joins(:foo).where(foos: { id: 1 })
+        Foo.left_joins(:foobar).where(foo: { id: nil })
+        Foo.left_joins(:foo).where(foobar: { id: nil })
+        Foo.left_joins(foo: :bar).where(bars: { id: nil })
```

Comment:
Can these be separate `it` ... `expect_no_offenses` to improve the independent execution of the test? The intent of each test code could also be clarified.

Reply:
Indeed it is. Thank you so much. I have split it into multiple it's depending on the interest of the test.